Source/WebCore/platform/graphics/mac/controls/ProgressBarMac.mm

```diff
@@ -126,6 +143,13 @@
 
     GraphicsContextStateSaver stateSaver(context);
 
+    if (isVerticalWritingMode) {
+        context.translate(inflatedRect.height(), 0);
+        context.translate(inflatedRect.location());
+        context.rotate(piOverTwoFloat);
+        context.translate(-inflatedRect.location());
+    }
```

Comment:
This is fine for now, but I assume we'll want some kind of shared helper so we don't duplicate this between different controls.

Reply:
Let's see, right now I think `<meter>` going to be the only other control which will benefit from this exact logic.